homeassistant/components/sonos/select.py

```diff
@@ -0,0 +1,131 @@
+"""Select entities for Sonos."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import (
+    ATTR_DIALOG_LEVEL,
+    ATTR_DIALOG_LEVEL_ENUM,
+    MODEL_SONOS_ARC_ULTRA,
+    SONOS_CREATE_SELECTS,
+)
+from .entity import SonosEntity
+from .helpers import SonosConfigEntry, soco_error
+from .speaker import SonosSpeaker
+
+
+@dataclass(frozen=True)
+class SonosSelectType:
+    """Data class for Sonos select types."""
+
+    feature: str
+    attribute: str
+    model: str
+    options: list[str]
+
+
+SELECT_TYPES: list[SonosSelectType] = [
+    SonosSelectType(
+        feature=ATTR_DIALOG_LEVEL,
+        attribute=ATTR_DIALOG_LEVEL_ENUM,
+        model=MODEL_SONOS_ARC_ULTRA,
+        options=["off", "low", "medium", "high", "max"],
+    ),
+]
```

Comment:
Why don't we use entity descriptions?

Reply:
Updated.  Was trying to use the existing pattern in the integration; instead of the more modern way...